### PR TITLE
Create TTS Engine for built-in macOS `say` command line tool

### PR DESCRIPTION
**Pull request by apfritts**

### CODE REVIEW

**Comment by apfritts** [comment]
Hi @grigi! I'm not sure what I did to cause the last test to fail. Would you mind taking a look? 💃

**Comment by grigi** [comment]
Hmm, looks like the SayTTS tests are failing. which is expected as `say` is an OSX tool, and doesn't exist on Linux (which travis uses for CI tests). You have the SKIP_IF_NOT_AVAILABLE set, so it should skip.
I don't have an OSX system to test on, so I'll just have to assume your code is working. I'll work on trying to figure out what is actually going on.

File: talkey/engines/say.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    def _get_init_options(cls):
+        return {
+            'say': {
+                'description': 'Say executable path',
+                'type': 'str',
+                'default': r'say'
+            },
+        }
+
+    def is_available(self):
```
Comment (grigi) [inline talkey/engines/say.py]: This should be `_is_available` else the abstract base complains.

**Comment by grigi** [comment]
hmm, it looks better, but is still failing. I however think it isn't your fault. I should probably drop py2.6 support as that is ancient by now.
And something caused a timeout... I suspect MaryTTS, the demo server is often down. Maybe make that auto skip if not available?
Either way, I don't think it is your fault.

**Comment by apfritts** [comment]
It looks like I required a package incompatible with 2.6 or something...I'll work on this later tonight :)

**Comment by grigi** [comment]
The last 2.6 build that passed was using gTTS 1.0.6, the newer 1.1 range seems to have dropped Py2.6 support, and I tested that build, it doesn't work anymore. I suspect some API change on googles side. I'd say we just drop support for 2.6.

**Comment by apfritts** [comment]
👍🏻👍🏻👍🏻 I can add that.

**Comment by grigi** [comment]
Ok, the one test failing is to do with the weighted language detection... Maybe another lib change?
I think your work is done, so I'll merge this now, and clean it up over the next week.
Thanks for your work :-)

**Comment by grigi** [comment]
Hi apfritts

Is the say engine working? I tried to set up travis to run tests on OSX (since I don't have access to such a system) and it is getting stuck at say() for say [sic].
https://travis-ci.org/grigi/talkey/jobs/146593671

Could you please post the output of running tests locally, and confirm what is going on?
It looks like running `say "blah"` causes it to hang.

**Comment by grigi** [comment]
Hi apfritts

Is the say engine working? I tried to set up travis to run tests on OSX (since I don't have access to such a system) and it is getting stuck at say() for say [sic].
https://travis-ci.org/grigi/talkey/jobs/146593671

Could you please post the output of running tests locally, and confirm what is going on?
It looks like running `say "blah"` causes it to hang.

**Comment by apfritts** [comment]
Yep! I'm not sure I have the tests setup on my MacBook correctly. I get 2 failures, 9 errors, and 22 skipped:
https://gist.github.com/apfritts/7660e3f0be4c531bbb7368303883e01d

I'm in the Canadian wilderness for a week. I'll play with it a little more but won't be able to check in until next week. It sounded correct (say worked) so I think the "say" tests are working correctly. However, given the number of failures on other tests that should pass, my confidence isn't high.

**Pull request merged**